### ISSUE TITLE
S3 event compile creates an extra bucket with the event configuration

### ISSUE THREAD

**Issue by bennybauer** [serverless/serverless#1823]
* Check if an Issue already exists for this
* Add the relevant label(s)
* Be kind :)

Bug Report Template -----------------------------------------------

##### Serverless Framework Version: 
v1.0-beta
##### Operating System: 
Mac OS X (10.11.6)
##### Expected Behavior:
S3 event should add an event on the configured bucket. 
For example, given the following `serverless.yml`:
```
functions:
  first:
      handler: handler.first
      events:
        - s3: mybucket

resources:
  Resources:
    newCoolResource:
      Type: AWS::S3::Bucket
      Properties:
        BucketName: mybucket
```
Should create the following in the CLF template:
```
"newCoolResource": {
      "Type": "AWS::S3::Bucket",
      "Properties": {
        "BucketName": "mybucket",
        "NotificationConfiguration": {
          "LambdaConfigurations": [{
            "Event": "s3:ObjectCreated:*",
            "Function": {
              "Fn::GetAtt": ["first", "Arn"]
            }
          }]
        }
      }
    },
```

##### Actual Behavior:
The generated CLF template has the following:
```
  "newCoolResource": {
      "Type": "AWS::S3::Bucket",
      "Properties": {
        "BucketName": "mybucket"
      }
    },

   "mybucket0S3Event": {
      "Type": "AWS::S3::Bucket",
      "Properties": {
        "BucketName": "mybucket0",
        "NotificationConfiguration": {
          "LambdaConfigurations": [{
            "Event": "s3:ObjectCreated:*",
            "Function": {
              "Fn::GetAtt": ["first", "Arn"]
            }
          }]
        }
      }
    },
```

The incorrect behaviors are: 
1. Two buckets were created: `mybucket` and `mybucket0` instead of only the first one.
2. The event was configured for `mybucket0` instead of the configured bucket.

**Comment by havard024** [comment]
Any traction on this bug? 

According to this: https://github.com/serverless/serverless/pull/1844 
reusing resources in events should be supported in 1.0.0-beta.2, but I still get the zero appended on the event bucket name even though I've created a bucket resource.

**Comment by flomotlik** [comment]
This will be resolved with #1869 so I'm closing here. Thanks for reporting this, it definitely needs to be resolved.
